qutebrowser/browser/greasemonkey.py

```diff
@@ -171,23 +194,110 @@ def load_scripts(self):
                 if not script.name:
                     script.name = script_filename
 
-                if script.run_at == 'document-start':
-                    self._run_start.append(script)
-                elif script.run_at == 'document-end':
-                    self._run_end.append(script)
-                elif script.run_at == 'document-idle':
-                    self._run_idle.append(script)
+                if script.requires:
+                    log.greasemonkey.debug(("Deferring script until "
+                                            "requirements are "
+                                            "fullfilled: {}")
```

Comment:
fullfilled -> [fulfilled](https://www.merriam-webster.com/dictionary/fulfill) - also, no need for the additional parens